lib/hls/manifest_text_parser.js

```diff
@@ -158,7 +158,13 @@ shaka.hls.ManifestTextParser = class {
             partialSegmentTags.push(tag);
           } else if (tag.name == 'EXT-X-PRELOAD-HINT') {
             if (tag.getAttributeValue('TYPE') == 'PART') {
-              partialSegmentTags.push(tag);
+              if (tag.getAttributeValue('BYTERANGE-START') != null) {
+                if (tag.getAttributeValue('BYTERANGE-LENGTH') != null) {
+                  partialSegmentTags.push(tag);
+                }
+              } else {
+                partialSegmentTags.push(tag);
+              }
```

Comment:
If you want to go straight to the recommendation of https://www.akamai.com/blog/performance/-using-ll-hls-with-byte-range-addressing-to-achieve-interoperabi, do this instead in `createSegmentReference_`:

```js
          // A preload hinted partial segment may have byterange length info.
          const pByterangeLength = item.getAttributeValue('BYTERANGE-LENGTH');
          if (pByterangeLength) {
            pEndByte = pStartByte + Number(pByterangeLength) - 1;
          }
          // *** ADD THIS ***
          else if (pStartByte) {
            // If we have a non-zero start byte, but no end byte, follow the recommendation of
            // https://tinyurl.com/hls-open-byte-range and set the end byte explicitly to a
            // large integer.
            pEndByte = Number.MAX_SAFE_INTEGER;
          }
```

I haven't tested that against the content in #5089.  If it doesn't work, we can just ignore the HINT segment instead.

Reply:
Yes, I have tried and thought about that, but the problem is that it overlaps with the following partial segments and very rare cases occur. That's why I created this solution.

Reply:
I think in the future we need to rethink the way we work with byterange, but this is a easy-fix in the meantime.

Reply:
How does MAX_SAFE_INTEGER on preload hint interact poorly with partial segments?

Reply:
Ah, I see.  It seems that Will Law's article suggests getting and using that continuation on purpose, but Shaka Player isn't built for it at the moment.